spring-security-oauth2/src/main/java/org/springframework/security/oauth2/config/annotation/web/configurers/AuthorizationServerSecurityConfigurer.java

```diff
@@ -128,17 +159,24 @@ public String getCheckTokenAccess() {
 
 	@Override
 	public void init(HttpSecurity http) throws Exception {
-
 		registerDefaultAuthenticationEntryPoint(http);
-		if (passwordEncoder != null) {
-			ClientDetailsUserDetailsService clientDetailsUserDetailsService = new ClientDetailsUserDetailsService(clientDetailsService());
-			clientDetailsUserDetailsService.setPasswordEncoder(passwordEncoder());
-			http.getSharedObject(AuthenticationManagerBuilder.class)
-					.userDetailsService(clientDetailsUserDetailsService)
-					.passwordEncoder(passwordEncoder());
+		AuthenticationManagerBuilder builder = http.getSharedObject(AuthenticationManagerBuilder.class);
+		if (authenticationEventPublisher != null) {
+		    builder.authenticationEventPublisher(authenticationEventPublisher);
 		}
-		else {
-			http.userDetailsService(new ClientDetailsUserDetailsService(clientDetailsService()));
+		if (authenticationProviders.isEmpty()) {
+			if (passwordEncoder != null) {
+				ClientDetailsUserDetailsService clientDetailsUserDetailsService = new ClientDetailsUserDetailsService(clientDetailsService());
+				clientDetailsUserDetailsService.setPasswordEncoder(passwordEncoder());
+				builder.userDetailsService(clientDetailsUserDetailsService)
+					.passwordEncoder(passwordEncoder());
+			} else {
+				builder.userDetailsService(new ClientDetailsUserDetailsService(clientDetailsService()));
+			}
+		} else { 
+			for(AuthenticationProvider provider: authenticationProviders) {
+				builder.authenticationProvider(provider);
```

Comment:
This additional configuration is not needed as you can register one or more `AuthenticationProvider`(s) via `HttpSecurity.authenticationProvider()`.